src/module/actor/creature/document.ts

```diff
@@ -393,8 +393,12 @@ abstract class CreaturePF2e<
         const { attributes, rollOptions } = this;
 
         // Add creature-specific self: roll options
-        if (this.isSpellcaster) {
+        const traditions = new Set(
```

Comment:
This actually looks a tad heavy. It'd pretty be lighter if don't from the spellcasting entries.

Reply:
You're last sentence doesn't parse for me.  Don't make a set?  It's possible to have the same spellcasting type from multiple entries.  I suppose it doesn't matter to set the same roll option multiple times.

Reply:
Sorry about that, swyping from couch!

Reply:
Do you mean to add this instead to `item.spellcasting-entry.document.preapreActorData()`?

Should the base `self:caster` option go there too? 

Like this:

```diff
--- a/src/module/item/spellcasting-entry/document.ts
+++ b/src/module/item/spellcasting-entry/document.ts
@@ -135,6 +135,11 @@ class SpellcastingEntryPF2e<TParent extends ActorPF2e | null = ActorPF2e | null>
                 4,
             ) as OneToFour;
         }
+
+        if (this.spells && this.spells.size > 0) {
+           actor.rollOptions.all["self:caster"] = true;
+           actor.rollOptions.all[`self:caster:${this.tradition}`] = true;
+        }
     }

     /** Prepares the statistic for this spellcasting entry */
```

Reply:
Yeah, like that. Use `self:caster:tradition:${this.tradition}` as well.